cranelift/isle/isle/isle_examples/pass/let_shadowing.isle

```diff
@@ -0,0 +1,38 @@
+
+(type u64 (primitive u64))
+
+(decl foo (u64) u64)
+(rule (foo x) x)
+
+;; Shadowing of a global name
+(decl test1 (u64) u64)
+(rule (test1 x)
+      (let ((foo u64 x))
+        foo))
+
+;; Shadowing of a parameter
+(decl test2 (u64) u64)
+(rule (test2 x)
+      (let ((x u64 x))
+        x))
+
+;; Shadowing of this binding's name
+(decl test3 (u64) u64)
+(rule (test3 x)
+      (let ((test3 u64 x))
+        test3))
+
+;; Shadowing another let-bound name
+(decl test4 (u64) u64)
+(rule (test4 x)
+      (let ((val u64 x)
+            (val u64 23))
```

Comment:
Would you mind turning at least this case into a run-test and invoking the rule from a test toplevel, asserting that the most recent binding wins?

Reply:
I reworked it as a `link` test in the `isle/isle_examples/link` directory, and modified the test runner to also execute the output of `rustc` on these tests.

Reply:
Ah, I think you may need to rebase -- as of Friday there is now a notion of a run-test already and a separate `isle_examples/run` directory :-)

Reply:
Nice! I rebased and removed the changes I introduced to run the `link` tests :+1: